File: db4on/reference/vb/db4o-reference-chapters/src/com/db4odoc/f1/refactoring/newclasses/RefactoringExample.vb
' Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com 
Imports System
Imports System.IO
Imports com.db4o
Imports com.db4o.query
Imports com.db4o.ext

Namespace com.db4odoc.f1.refactoring.Newclasses
    Public Class RefactoringExample
        Public Shared ReadOnly YapFileName As String = "formula1.yap"

        Public Shared Sub Main(ByVal args() As String)
            ReopenDB()
            TransferValues()
        End Sub
        ' end Main

        Public Shared Sub ReopenDB()
            Db4o.Configure().DetectSchemaChanges(False)
            Dim oc As ObjectContainer = Db4o.OpenFile(YapFileName)
            oc.Close()
        End Sub
        ' end ReopenDB

        Public Shared Sub TransferValues()
            Dim oc As ObjectContainer = Db4o.OpenFile(YapFileName)
            Try
                Dim sc As StoredClass = oc.Ext().StoredClass(GetType(Pilot))
                System.Console.WriteLine("Stored class:  " + sc.GetName())
                Dim sfOld As StoredField = sc.StoredField("_name", GetType(String))
                System.Console.WriteLine("Old field:  " + sfOld.GetName() + ";" + sfOld.GetStoredType().GetName())
                Dim q As Query = oc.Query()
                q.Constrain(GetType(Pilot))
                Dim result As ObjectSet = q.Execute()
                Dim i As Integer
                For i = 0 To result.Size() - 1 Step i + 1
                    Dim pilot As Pilot = CType(result(i), Pilot)
                    pilot.Name = New Identity(sfOld.Get(pilot).ToString(), "")
                    System.Console.WriteLine("Pilot=" + pilot.ToString())
                    oc.Set(pilot)
                Next
            Finally
                oc.Close()
            End Try
        End Sub
        ' end TransferValues
    End Class
End Namespace

File: db4on/reference/db4o-reference-chapters/src/com/db4odoc/f1/refactoring/newclasses/RefactoringExample.cs
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */
using System;
using System.IO;
using com.db4o;
using com.db4o.query;
using com.db4o.ext;

namespace com.db4odoc.f1.refactoring.newclasses
{
	public class RefactoringExample
	{
		public readonly static string YapFileName = "formula1.yap";
	
		public static void Main(string[] args) 
		{
			ReopenDB();
			TransferValues();
		}
		// end Main

		public static void ReopenDB()
		{
			Db4o.Configure().DetectSchemaChanges(false);
			ObjectContainer oc = Db4o.OpenFile(YapFileName);
			oc.Close();
		}
		// end ReopenDB

		public static void TransferValues()
		{
			ObjectContainer oc = Db4o.OpenFile(YapFileName);
			try 
			{
				StoredClass sc = oc.Ext().StoredClass(typeof(Pilot));
				System.Console.WriteLine("Stored class:  "+ sc.ToString());
				StoredField sfOld = sc.StoredField("_name",typeof(string));
				System.Console.WriteLine("Old field:  "+ sfOld.ToString()+";"+sfOld.GetStoredType());
				Query q = oc.Query();
				q.Constrain(typeof(Pilot));
				ObjectSet result = q.Execute();
				for (int i = 0; i< result.Size(); i++)
				{
					Pilot pilot = (Pilot)result[i];
					pilot.Name = new Identity(sfOld.Get(pilot).ToString(),"");
					System.Console.WriteLine("Pilot="+ pilot);
					oc.Set(pilot);
				}
			} 
			finally 
			{
				oc.Close();
			}
		}
		// end TransferValues
	}
}
